### PR TITLE
Create New Paths for Dev_USB and mms

### PR DESCRIPTION
**Pull request by Darkhost1999**
I noticed that RPCS3 could use some more PS3 file structures.
Main intention is to aid converting savedata for usb users of Apollo Save Tool [NP0APOLLO]

### CODE REVIEW

File: rpcs3/Emu/Io/usio.cpp [hunk 1]
```diff
@@ -122,12 +122,10 @@ void usb_device_usio::control_transfer(u8 bmRequestType, u8 bRequest, u16 wValue
 	transfer->fake = true;
 
 	// Control transfers are nearly instant
-	switch (bmRequestType)
```
Comment (elad335) [inline rpcs3/Emu/Io/usio.cpp:125]: Just comment it out.

File: rpcs3/Emu/System.cpp [hunk 2]
```diff
@@ -424,7 +424,18 @@ void Emulator::Init()
...
+		if (make_path_verbose(dev_usb, true))
+		{
+			make_path_verbose(dev_usb + "MUSIC/", false);
+			make_path_verbose(dev_usb + "VIDEO/", false);
+			make_path_verbose(dev_usb + "PICTURE/", false);
+			make_path_verbose(dev_usb + "PS3/EXPORT/PSV/", false); // PS1 and PS2 Saves go here
+			make_path_verbose(dev_usb + "PS3/SAVEDATA", false);
+			make_path_verbose(dev_usb + "PS3/THEME", false);
+			make_path_verbose(dev_usb + "PS3/UPDATE", false);
+
```
Comment (Megamouse) [inline rpcs3/Emu/System.cpp:437]: empty line

File: rpcs3/util/dyn_lib.hpp [hunk 3]
```diff
@@ -14,15 +14,15 @@ namespace utils
 
 		dynamic_library(const dynamic_library&) = delete;
 
-		dynamic_library(dynamic_library&& other)
+		dynamic_library(dynamic_library&& other) noexcept
```
[the remark this answers is not in the document]
Comment (Darkhost1999) [reply rpcs3/util/dyn_lib.hpp:17]: I originally resolved the warning on June 11th. I should've made better notes on what warning.

File: rpcs3/Emu/System.cpp [hunk 4]
```diff
@@ -453,6 +463,9 @@ void Emulator::Init()
 			make_path_verbose(dev_hdd0 + "savedata/", false);
 			make_path_verbose(dev_hdd0 + "savedata/vmc/", false);
 			make_path_verbose(dev_hdd0 + "photo/", false);
+			make_path_verbose(dev_hdd0 + "music/", false);
+			make_path_verbose(dev_hdd0 + "theme/", false);
+			make_path_verbose(dev_hdd0 + "video/", false);
```
[the remark this answers is not in the document]
Comment (Darkhost1999) [reply rpcs3/Emu/System.cpp:468]: I used this website
https://www.psdevwiki.com/ps3/Files_on_the_PS3
And what I was able to confirm from category_music.xml, category_video.xml, category_photo.xml, and the other files while also testing functionality to see that these paths work on real ps3 and in rpcs3.
I was not able to find a dev_hdd0 file structure so well spelled out as dev_usb.
And there were paths such as otheros that I left out of implementing.
Do you know of a resource that spells out what the PS3's file structure is of OFW for dev_hdd0?

[the remark this answers is not in the document]
Comment (elad335) [reply rpcs3/Emu/System.cpp:468]: /drm, /vsh, /crash_report, /game_debug

[the remark this answers is not in the document]
Comment (elad335) [reply rpcs3/Emu/System.cpp:468]: I haven't found the additional directory mega was talkiing about on DECR, maybe it is created when calling cellPhotoImport.

[the remark this answers is not in the document]
Comment (Megamouse) [reply rpcs3/Emu/System.cpp:468]: Yes. It's created when you use the weird LBP devtool for cellPhotoImport

**Pull request merged**